packages/web-app/src/modules/profile/ProfileStore.ts

```diff
@@ -76,4 +76,33 @@ export class ProfileStore {
       // this.store.routing.replace('/')
     }
   })
+
+  @action.bound
+  updateMinecraftUsername = flow(function* (this: ProfileStore, minecraftUsername: string) {
+    if (this.currentProfile === undefined) {
+      return
+    }
+
+    this.isUpdating = true
+
+    try {
+      let patch = yield this.axios.patch('/api/v1/profile', {
+        extensions: {
+          minecraftUsername: minecraftUsername,
+        },
```

Comment:
The `PATCH` method is cool with merging in changes to top-level properties... but it doesn't preserve any child properties. For the most part, this is why we don't frequently use nested objects. We need to explicitly preserve any other extension properties, otherwise this will be _deleting_ the others.

```suggestion
        extensions: this.currentProfile.extensions === undefined
          ? {
              minecraftUsername: minecraftUsername,
            }
          : {
              ...this.currentProfile.extensions,
              minecraftUsername: minecraftUsername,
            },
```